src/test/java/io/debezium/connector/spanner/DataTypesIT.java

```diff
@@ -0,0 +1,106 @@
+    private static final List<String> types = Arrays.asList("BOOL", "INT64",
+            "FLOAT32", "FLOAT64", "TIMESTAMP", "DATE", "STRING", "BYTES",
+            "NUMERIC", "JSON");
+
+    @BeforeAll
+    static void setup() throws InterruptedException, ExecutionException {
+
+        databaseConnection.createTable(tableName + "(id INT64,"
+                + "  boolCol BOOL,"
+                + "  int64Col INT64,"
+                + "  float32Col FLOAT32,"
+                + "  float64Col FLOAT64,"
+                + "  timestampCol TIMESTAMP,"
+                + "  dateCol DATE,"
+                + "  stringCol STRING(MAX),"
+                + "  bytesCol BYTES(MAX),"
+                + "  numericCol NUMERIC,"
+                + "  jsonCol JSON,"
+                + "  arrCol ARRAY<STRING(MAX)>,"
+                + ") PRIMARY KEY (id)");
+        databaseConnection.createChangeStream(changeStreamName, tableName);
+
+        System.out.println("DataTypesIT is ready...");
+    }
+
+    @AfterAll
+    static void clear() throws InterruptedException {
+        databaseConnection.dropChangeStream(changeStreamName);
+        databaseConnection.dropTable(tableName);
+    }
+
+    @Test
+    public void shouldStreamUpdatesToKafkaWithTheCorrectType() throws InterruptedException {
+        final Configuration config = Configuration.copy(baseConfig)
+                .with("gcp.spanner.change.stream", changeStreamName)
+                .with("name", tableName + "_test")
+                .with("gcp.spanner.start.time",
+                        DateTimeFormatter.ISO_INSTANT.format(Instant.now()))
+                .build();
+        initializeConnectorTestFramework();
+        start(SpannerConnector.class, config);
+        assertConnectorIsRunning();
+        databaseConnection.executeUpdate("INSERT INTO " + tableName
+                + "(id, boolCol, int64Col, float32Col, float64Col, timestampCol,"
+                + " dateCol, stringCol, bytesCol, numericCol, jsonCol, arrCol) "
+                + "VALUES"
+                + " (1, true, 42, 3.14, 2.71, '1970-01-01 00:00:00 UTC',"
+                + "  '1970-01-01', 'stringVal', b'bytesVal', 6.023,"
+                + "  JSON '\"Hello\"', ['a', 'b'])");
+        waitForAvailableRecords(waitTimeForRecords(), TimeUnit.SECONDS);
+        SourceRecords sourceRecords = consumeRecordsByTopic(10, false);
+        List<SourceRecord> records = sourceRecords.recordsForTopic(getTopicName(config, tableName));
+        assertThat(records).hasSize(1);
+
+        Struct record = (Struct) (records.get(0).value());
+        assertThat(record.get("op")).isEqualTo("c");
+        assertThat(record.schema().field("after")).isNotNull();
+
+        Struct values = record.getStruct("after");
+
+        assertTrue(values.getBoolean("boolCol"));
+        assertThat(values.getInt64("int64Col")).isEqualTo(42);
+        // TODO: uncomment after emulator is released with the FLOAT32 support.
```

Comment:
Two things:

(1) Will you submit followup PR to uncomment this line?

(2) Will you submit followup PR to add Postgres IT?

Reply:
Thanks!

1) Yes, I'm waiting for the next emulator release which will have the fix. I'll uncomment after that.

2) I've tried and failed to add the postgres dialect IT in this PR (attempt: https://github.com/arawind/debezium-connector-spanner/pull/1); so yes I'll follow up after I get it to work. I couldn't find any existing integration tests with PG dialect, so I'm not sure if there are more changes needed in the IT framework or if I'm doing something wrong.

Reply:
I'm working with @ShuranZhang to fix the PG integration test. Since I'm unable to get BasicSanityIT pass with PG dialect, I assume there are unrelated changes that need to be done before I can add the DataTypesIT for PG dialect.

Given this, I would prefer if we can go ahead with this PR and handle the changes in a follow up.

Reply:
Hi @arawind ,

I just checked your draft pr for adding PG it test and noticed a problem: 
![image](https://github.com/user-attachments/assets/5058db72-26b2-49a5-bcd7-e9f19950e372)

you directly copied [baseConfig](https://github.com/debezium/debezium-connector-spanner/blob/main/src/test/java/io/debezium/connector/spanner/AbstractSpannerConnectorIT.java#L34) when running your IT test connectors. However this `baseConfig` is hardcoded to connect to the pre-created googlesql database with `database.getDatabaseId()`. It can't reach the `pgDatabase` you defined in-place in `DataTypesIT.java`. For this test case under PG dialect, the connector will still connects to the gsql database successfully and thus receives no records when you write to PG dialect. You might need to modify the test util files and `AbstractSpannerConnectorIT` to include a pre-created pg database and make the desired database in baseConfig configurable.

I would also recommend to use different table names for different dialects to be more clear.

Reply:
Thanks @ShuranZhang for your offline help!  I was able to get the PG integration tests working now after making all the names to lowercase for PG.

A couple of updates and findings: 
1. I could not parameterize the tests for PG and GoogleSQL, as the test framework doesn't support it. I had to create a separate test class for PG.
2. FLOAT32 in changestreams is now supported in the GoogleSQL dialect, so that test class does not have any tech-debt/TODOs remaining.
3. FLOAT32 as a whole isn't yet supported in PG dialect of emulator, so I've added a TODO for that and commented out the columns in the test. I've verified with the internal version that the test works.
4. I found through the integration test, that NUMERIC and JSONB aren't supported in the PG dialect of change streams for some reason. Could someone dig into this? I've commented out those columns too.

@nancyxu123 could you please review once again? #1 has been addressed, and #2 is partially done.

Reply:
For #4: JSONB and NUMERIC should be supported for change streams in Emulator and we have integration tests continuously running without issue forJSONB and NUMERIC in Emulator repo. I suspect this issue is potentially on the debezium connector end. I will create a tracking bug and investigate it. 

cc: @nancyxu123

Reply:
Thanks @ShuranZhang! Resolving this discussion for this PR.